features/step_definitions/redmine_steps.rb

```diff
@@ -7,59 +7,68 @@
 end
 
 Then(/^New user is registered$/) do
-  success_message = find_element_by_id('flash_notice')
 
-  expect(@driver.current_url).to eql @my_account_page_url
-  expect(success_message).to be_displayed
+
+  success_message = on(MyAccountPage).success_message_element
+
+  expect(@current_page.page_url_value).to eql on(MyAccountPage).page_url_value
+  expect(success_message).to be_visible
 end
 
 Then(/^I am logged in$/) do
-  expect(find_element_by_class('user').text).to eql @user.login
+  active_username = on(MyPage).top_menu.active_user_element.attribute 'text'
+  expect(active_username).to eql @user.login
 end
 
 When(/^I log out$/) do
-  log_out
+  on(TopMenuSection).log_out
 end
 
 
 Then(/^I am logged out$/) do
-  login_button = find_element_by_class('login')
-  expect(@driver.current_url).to eql @homepage_url
-  expect(login_button).to be_displayed
+  login_button = on(HomePage).top_menu.log_in_element
+
+  expect(@current_page.page_url_value).to eql on(HomePage).page_url_value
+  expect(login_button).to be_visible
 end
 
 
 Given(/^I register a user$/) do
-  register_user(@user)
+  visit(RegistrationPage).register_user(@user)
 end
 
 And(/^I log in$/) do
-  log_in(@user.login, @user.password)
+  visit(LoginPage).log_in(@user)
 end
 
 When(/^I change password$/) do
-  @user.password = change_password(@user)
+  new_password = Faker::Internet.password
+  visit(ChangePasswordPage).change_password(@user.password, new_password)
+
+  @user.password = new_password
 end
 
 Then(/^Success message displayed$/) do
-  success_message = find_element_by_id('flash_notice')
-  expect(success_message).to be_displayed
+  success_message = on(MyAccountPage).success_message_element
+  expect(success_message).to be_visible
 end
 
 
 And(/^I can login with a new password$/) do
-  log_out
-  log_in(@user.login, @user.password)
+  step 'I log out'
```

Comment:
Refactored step_definitions and removed `step` usages.

Btw I was looking for a workaround (how to reuse already defined steps in step definitions to keep them DRY) and found this: https://makandracards.com/makandra/18905-how-to-not-repeat-yourself-in-cucumber-scenarios

There's a `many_steps.rb` helper proposed by author so it supposed to solve problem:
> When calling steps with multiple lines of Cucumber you lose meaningful stack traces. If a step fails you will always get the same file and line number: The one where you call steps. This is a big deal in practice. To remedy this, check out our many_steps helper (see below).

What do you think about this practice, maybe there's any graceful way to avoid code duplication if i need to gather some steps within one to keep scenario simple?